### PR TITLE
feat: update link to legacy curriculum maps

### PR DESCRIPTION
**Pull request by lioneljingles**
## Description

Music year: {owa_music_year}

- Change legacy curriculum link in Subject Phase Picker to point previous downloads page

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2250--oak-web-application.netlify.thenational.academy/teachers/curriculum
2. Click on `Subject` control
3. Click on `Previously Released Plans`
4. You should navigate to the "Previously released curricula" page with the path `/teachers/curriculum/previous-downloads`

## Checklist

- [x] Added or updated tests where appropriate
- [x] Manually tested across browsers / devices
- [x] Considered impact on accessibility
- [x] Design sign-off
- [x] Approved by product owner
- [x] Does this PR update a package with a breaking change

### CODE REVIEW

**Review by rightsaidjames — APPROVED** [review]
Approved, assuming that either the above change is made or you have a good reason not to make it!

File: src/components/SharedComponents/SubjectPhasePicker/SubjectPhasePicker.tsx [hunk 1]
```diff
@@ -448,9 +448,10 @@ const SubjectPhasePicker: FC<SubjectPhasePickerData> = ({
...
                 <Box $mt={24}>
                   <OwaLink
-                    page={"oak-curriculum"}
+                    page={"curriculum-previous-downloads"}
                     $textDecoration={"underline"}
                     $font={"heading-7"}
+                    data-testid="previousPlansLink"
                   >
                     Previously released plans
                     <Icon
```
Comment (rightsaidjames) [inline src/components/SharedComponents/SubjectPhasePicker/SubjectPhasePicker.tsx:457]: @HelenHarlow Can you confirm what icon you would like this changed to?

Comment (HelenHarlow) [reply src/components/SharedComponents/SubjectPhasePicker/SubjectPhasePicker.tsx:457]: Could we use an arrow please?
Screenshot 2024-02-20 at 16 29 10

Comment (rightsaidjames) [reply src/components/SharedComponents/SubjectPhasePicker/SubjectPhasePicker.tsx:457]: @lioneljingles Over to you!

**Review by rightsaidjames — CHANGES_REQUESTED** [review]
As above, just the icon change needed and we're good to go!

**Review by rightsaidjames — APPROVED** [review]
All good!

**Pull request merged**